pkg/api/resource/quantity.go

```diff
@@ -385,6 +385,15 @@ func ParseQuantity(str string) (Quantity, error) {
 	return Quantity{d: infDecAmount{amount}, Format: format}, nil
 }
 
+// DeepCopy returns a deep-copy of the Quantity value.
+func (q Quantity) DeepCopy() Quantity {
+	if q.d.Dec != nil {
+		tmp := &inf.Dec{}
+		q.d.Dec = tmp.Set(q.d.Dec)
+	}
+	return q
```

Comment:
Hmm - I don't understand it. Shouldn't we have a line like:
result := q
at the beginning?

How this works?